src/core/retargeting_engine/python/interface/tensor_group.py

```
@@ -93,3 +119,52 @@ def get_tensor(self, index: int) -> Tensor:
+        return otg
+
+    @property
+    def is_none(self) -> bool:
+        """Whether this group is in the absent state."""
+        return self._is_none
+
+    def set_none(self) -> None:
+        """Mark this group as absent (no data available)."""
+        self._is_none = True
+
+    def __getitem__(self, index: int) -> Any:
```

Ruff (0.15.2)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `__getitem__`

(ANN401)